plugins/sigma/src/parse.cpp

```diff
@@ -316,33 +323,51 @@ caf::expected<expression> parse_search_id(const data& yaml) {
         } else if (*i == "utf16") {
           return caf::make_error(ec::unimplemented, //
                                  "utf16 not yet implemented");
-        } else if (*i == "endswith" || *i == "startswith" || *i == "re") {
+        } else if (*i == "startswith") {
+          auto to_re = [](const data& d) -> caf::expected<data> {
+            auto f = detail::overload{[](const auto& x) -> caf::expected<data> {
+              auto str = fmt::format("^{}.*", to_string(x));
+              auto result = pattern::make(str);
+              if (!result)
+                return std::move(result.error());
+              return std::move(*result);
```

Comment:
The early return in case of an error is much more common throughout the code base, actually, at least in code added in the last year or so.

An NB on your NB: We most definitely should be moving the result of the error function on an expected as that returns by lvalue ref and errors are pretty expensive to copy. So much so, that a missing move in a hot loop once caused the summarize operators to be 20% slower than it could've been in a code pattern just like this.